Recipes to help limit hopper + farm storage chests

Convoy20 opened the pull request:
These recipes are designed to reduce the reliance on bots for new players and to encourage more compact output storage for farms. Diet added this config onto civclassics test and I've gone through and made sure it works. (This change also includes the craft_paper recipe which is not on test yet)

On the farm front: if you produce 10,000 golden nuggets over the course of a couple days, you need 3 doublechests worth of storage, since that is 156 stacks of nuggets. Now if you turn them into ingots in a gold forge factory, you now only have 17 stacks of ingots, you could fit that in a single chest. Taking that even further and condensing it down to gold blocks, you barely need two slots to store all the gold blocks from your farm's production. This will reduce the need for storage chests at gold farms by 98.77%, reducing the need for more hoppers and reducing the need for excessive minecart chests.

Added these recipes:

smelt_stone_compacted - REMOVED
smelt_glass_compacted  - REMOVED
craft_wool
craft_clay
craft_paper
craft_glass_bottle
bamboo_to_stick
make_bone_meal
make_bone_block
make_piston
forge_gold_ingot
forge_gold_block
+ modified gold forge hoe recipe to make probabilities match up to 100% (removes a factorymod warning)

The two compacted recipes are pretty self explanatory, they use the same amount of charcoal + take the same amount of time as it would take to run on uncompacted equivalents. - **REMOVED** both recipes, will add as a different PR if grinding mill stuff goes through since it seems controversial.

Craft wool is an optional recipe for anyone who wants to use a spider spawner for aesthetics instead of for obby production. 

Craft paper is useful if someone is creating a lot of printed notes or books.

Water bottle recipe gives vanilla outputs of bottles for glass - This is mainly in place for people who don't want to have to use a bot to make glass bottles for xp production.

Bone meal and bone block recipes can reduce storage needed for skeleton farms

Added piston to redstone mechanics factory since it wasn't there

Convoy20 committed:
Recipes to help limit hopper + farm storage chests

These recipes are designed to reduce the reliance on bots for new players and to encourage more compact output storage for farms. Diet added this config onto civclassics test and I've gone through and made sure it works. (This change also includes the craft_paper recipe which is not on test yet)

On the farm front: if you produce 10,000 golden nuggets over the course of a couple days, you need 3 doublechests worth of storage, since that is 156 stacks of nuggets. Now if you turn them into ingots in a gold forge factory, you now only have 17 stacks of ingots, you could fit that in a single chest. Taking that even further and condensing it down to gold blocks, you barely need two slots to store all the gold blocks from your farm's production. This will reduce the need for storage chests at gold farms by 98.77%, reducing the need for more hoppers and reducing the need for excessive minecart chests.

Added these recipes:

smelt_stone_compacted
smelt_glass_compacted
craft_wool
craft_clay
craft_paper
craft_glass_bottle
bamboo_to_stick
make_bone_meal
make_bone_block
make_piston
forge_gold_ingot
forge_gold_block
+ modified gold forge hoe recipe to make probabilities match up to 100% (removes a factorymod warning)

The two compacted recipes are pretty self explanatory, they use the same amount of charcoal + take the same amount of time as it would take to run on uncompacted equivalents. 

Craft wool is an optional recipe for anyone who wants to use a spider spawner for aesthetics instead of for obby production. 

Craft paper is useful if someone is creating a lot of printed notes or books.

Water bottle recipe gives vanilla outputs of bottles for glass - This is mainly in place for people who don't want to have to use a bot to make glass bottles for xp production.

Bone meal and bone block recipes can reduce storage needed for skeleton farms

Added piston to redstone mechanics factory since it wasn't there

Protonull commented:
Thinly veiled attempt to reduce iron costs while you fund your vault

Convoy20 commented:
> Thinly veiled attempt to reduce iron costs while you fund your vault

Is this comment directed at 

>smelt_stone_compacted
>smelt_glass_compacted

or at all the recipes? If you want me to continue doing what I do right now **which is not use crates** and spam more hoppers and doublechests everywhere I can gleefully so do.

Protonull commented:
> or at all the recipes? If you want me to continue doing what I do right now **which is not use crates** and spam more hoppers and doublechests everywhere I can gleefully so do.

👍

Convoy20 committed:
Remove compacted recipes - will wait to see how grinding mill changes go

Remove compacted recipes - will wait to see how grinding mill changes go

Convoy20 commented:
I removed both compacted recipes since people apparently dislike them.

Convoy20 commented:
(Things to think about adding in future, potentially - redstone block crafting/iron block crafting)

Protonull commented:
Something to keep in mind is that these proposed changes aren't just about adding convenient factory recipes for small scale automation or adding an alternative to botting as suggested by the given example, but rather to facilitate production on a truly enormous scale.. so not so much 10,000 nuggets but 200,000 nuggets, and thus he's balancing (or going to balance) the costs and outputs around such extremely high scale production.. so if the costs and outputs are not worth it for him and his use case then, using his own words, there's no point adding these recipes at all.

Convoy20 committed:
update recipes to better match vanilla equivalents - equalize coal cost to comparator 

update recipes to better match vanilla equivalents - equalize coal cost to comparator

Convoy20 commented:
> Something to keep in mind is that these proposed changes aren't just about adding convenient factory recipes for small scale automation or adding an alternative to botting as suggested by the given example, but rather to facilitate production on a truly enormous scale.. so not so much 10,000 nuggets but 200,000 nuggets, and thus he's balancing (or going to balance) the costs and outputs around such extremely high scale production.. so if the costs and outputs are not worth it for him and his use case then, using his own words, there's no point adding these recipes at all.

Thank you for taking an hour+ long conversation entirely out of context to strawman a point. 

I've modified the recipes that do not increase output to be equivalent coal cost to that of a comparator, to make them reasonable for a high outputting farm. The reasons being, yet again, that if the costs are too high, high outputting farms will simply rely on more overflow (50+ doublechests and hoppers, or minecart chests) - or just use a bot to avoid ridiculously high charcoal costs for something that ends up saving server tps usage.

Something to keep in mind is during our conversation you pretty much point blank said you would prefer factorymod didn't even exist (except for xp), so your feedback at this point is just petty and self serving for your own playstyle instead of balancing for what the server should be built around.

Quinndolin commented:
> 
> 
> Something to keep in mind is that these proposed changes aren't just about adding convenient factory recipes for small scale automation or adding an alternative to botting as suggested by the given example, but rather to facilitate production on a truly enormous scale.. so not so much 10,000 nuggets but 200,000 nuggets, and thus he's balancing (or going to balance) the costs and outputs around such extremely high scale production.. so if the costs and outputs are not worth it for him and his use case then, using his own words, there's no point adding these recipes at all.

Pull request to nerf portal farms then. Pretty irrelevant to whether these changes are a good idea or not. Convoy proposing these changes is also completely irrelevant to whether they're a good idea or not. High yield farms already exist, lets make them less laggy.

Protonull commented:
> you pretty much point blank said you would prefer factorymod didn't even exist (except for xp)

Yeah no, that's utter fake news.. you tried saying that in VC too and I rebuked it then too

Convoy20 commented:
> Yeah no, that's utter fake news.. you tried saying that in VC too and I rebuked it then too

Your view of factorymod is something to replace vanilla recipes that requires occasional upkeep to make it not attractive to use otherwise. Not for actual trade, production, industrialization of industries and the tech tree. Civ isn't a vanilla SMP.
 

>How is opposing your changes self serving towards my own playstyle? Genuinely perplexed as to where you got that idea from.. like I never, not once, said during VC that these changes will effect me or my playstyle.. just.. like.. what? I can only assume this is projection since this PR is basically for you and your playstyle, so you think my opposition to it is in the same vein.

Because you only see the game from your own viewpoint, and even during our VC you bitterly complain about how long it takes to travel across the map at 13 TPS. I introduce this PR specifically with not only my own high output farms in mind but others' as well who far outpace my own production, and you bitterly complain against it because you as a hermit won't benefit from it... when even though you do, when the TPS isn't being sucked up by high production farms.

These farms exist. More of them are being built. Refusing to add more things to factorymod to improve tps is just stupid.

Protonull commented:
> Your view of factorymod is something to replace vanilla recipes that requires occasional upkeep to make it not attractive to use otherwise. Not for actual trade, production, industrialization of industries and the tech tree. Civ isn't a vanilla SMP.

More fake news… I said that I think FactoryMod primarily creates economies of scale, where, like in real life, you need to capital to build and maintain a factory which can produce things more efficiently than you could otherwise, but you need to decide whether to invest that capital. It's true that the Aesthetics Factory for example doesn't abide by this, and I have mixed feelings about such factories and their recipes. But I have never *ever* supported FactoryMod being used to "replace" vanilla features, indeed I explicitly mentioned this as one of the failings of CivEx, and I even suggested a low-yield crafting recipe for XP just so that it's not a FactoryMod-exclusive thing. Again, you're just projecting because this PR is a bunch of vanilla recipe replacements.

----

> Because you only see the game from your own viewpoint, and even during our VC you bitterly complain about how long it takes to travel across the map at 13 TPS.

Uuuu*uuuuuh*, bitterly complained? What? You mean the casual mention of TPS lag as being a factor against looking for shops in nearby nations for items I don't mind crafting myself? You mean the conversation which ended in us both commiserating about the annoyances of trying to find stocked shops in other nations? Okay buddy, that was totally a bitter complaint.

----

> you bitterly complain against it because you as a hermit won't benefit from it

😨  You've found me out! It's like when I spent the last year advocating for keeping phantoms.. it's because I just got so much out of being randomly attacked and knocked off things. And why I advocated for disabling ice boating, because I get so much out of getting places slower and with much less food. And why I advocate for reducing Grinding Mill yields despite building building a GOR connection which uses gravel as its primary material. And why I advocated against ItemExchange relay recursion, because I get so much out of my shops being more vulnerable than they could be otherwise. And why I added item frame and lectern protection in Citadel because I'm infamous for my prolific use of such things.

…*oh wait*

----

Sooo can you please stop being so assblasted and making up lies about me, please? Nothing about [my comment](https://github.com/CivClassic/AnsibleSetup/pull/217#issuecomment-864649050) was incorrect:

* You *did* make this PR primarily for "high outputting farming"
* You *did* say that you're perfectly okay with botting instead
* You *did* say the example was lower than your use case
* You *did* say that if the charcoal cost is too high, there'd be no point in adding these recipes and you'd just bot anyway

Like seriously dude, you complain that my comment was out of context.. but what exactly about the comment is wrong?

Convoy20 commented:
> * You _did_ make this PR primarily for "high outputting farming"

To reduce impact on TPS. Not to improve my own farms or make them more optimal, **like you continually suggest**

> * You _did_ say that you're perfectly okay with botting instead

And that is what I will do, alongside way more laggy, absolutely tps destructive methods to make said botting viable. 

> * You _did_ say the example (10,000) was lower than your use case (200,000)

I used 200,000 in our VC as an example number, if I was to increase the number of portal farms I currently own and operate. I am running barely 1 portal farm right now. Not even close to the 200,000 number, but planning on reaching that number in the future is a damn good idea.

> * You _did_ say that if the charcoal cost is too high, there'd be no point in adding these recipes and you'd just bot anyway

Yes. Use your brain. If the cost for using these factories to reduce absolutely tps destructive methods is too expensive... will anyone use them?

> Like seriously dude, you complain that my comment was out of context.. but what exactly about the comment is wrong?

See above? I don't even feel like arguing about all your other wall of text while you're on your high horse.

Convoy20 commented:
My words: Your view of factorymod is something to replace vanilla recipes that requires occasional upkeep to make it not attractive to use otherwise.

>you need to **capital** to build and maintain a factory which can produce things more efficiently than you could otherwise, but you need to decide whether to invest that capital.... But I have never ever supported FactoryMod being used to "replace" vanilla features

Currently almost every factorymod recipe is replacement for vanilla recipes with improvements, and with capital costs. 

Like I'm confused here, do you want the factorymod recipes included in this PR to increase output of materials? Because that sounds *actually* beneficial to me??

Protonull commented:
> To reduce impact on TPS.

Yeeaaah, I still don't get why adding a bunch of FactoryMod recipes is preferable to adding pipes if your sole (or at least primary) argument is about TPS. (And before you try this again: I wont benefit from pipes) And I even suggested having iced flowing water and bubble column water paths instead of hopper chains.

----

> Not to improve my own farms or make them more optimal, **like you continually suggest**

Nope, that's another thing you've decided to read into my words.. which is seemingly becoming a habit. I said these PR changes are "to facilitate production on a truly enormous scale" and "he's balancing (or going to balance) the costs and outputs around such extremely high scale production." Both of these statements are true.. you rejected my cost:yield suggestion after you did some calculations based on your 200,000 theoretical number because you consider 3-4 DCs of charcoal too high for converting 78 DCs of materials. [I see you've gone through with your own cost:yield suggestions based on that 200,000 number.](https://github.com/CivClassic/AnsibleSetup/pull/217/commits/ffade9dcb2b8a76e74fbf8b06b0fb70ddddf9494)

----

> I used 200,000 in our VC as an example number, a number that is not current but if I was to increase the number of portal farms I currently own and operate (by a huge multiplier). I am running barely 1 portal farm right now. Not even close to the 200,000 number, but planning balancing choices around that number (which is not hard to reach) in the future is a damn good idea.

Don't think it was just an example, iirc it was an estimation of what you'd get if you activated all your portal farms / built all your planned portal farms. I'm inclined to believe the former since you said it took you three hours to convert the gold nugget yield. But either way, I never actually said you're producing 200,000… I said this PR is to facilitate production of 200,000, which is your use case.. which is true, you plan on doing this if you can.

----

> If the cost for using these factories to reduce absolutely tps destructive methods is too expensive... will anyone use them?

Well that depends on whether the recipes produce more output than they would with their equivalent crafting recipes. I've advocated against higher outputs because their stated intention (with the exception of the piston recipe) is to add automation, not add economies of scale. And so naturally the people who'll use those recipes are people seeking automation, but you've constantly argued on the premise that the only people who'll seek this automation is people handling 200,000-ish outputs.. I disagree. I accept that I'm probably naïve here and that 3-4 DCs of charcoal to process 78 DCs of material is too high.. but your proposed changes reducing it down to iirc like 1 DC is far too inexpensive. It's still FactoryMod.. you should still have to consider the costs of building, using, and maintaining the factory… and if you're casually handling 78 DCs of material of something, you can probably afford less than a full stack of compacted charcoal (produced from a bottable renewable resource) without much of an issue at all, which defeats the point.

----

> I don't even feel like arguing about all your other wall of text while you're on your high horse.

Sorry, but I'm not going to sit idly and smiley while you spread blatant lies about me.

----

> Like I'm confused here

The confusion comes because you're using "replacement" to mean "having a better version of", whereas I'm arguing with the truest sense of the word. I've consistently argued against FactoryMod *replacing* vanilla mechanics, like how CivEx *replaced* vanilla item repairs with a custom system, same with potions. Like, I'm really not losing sleep at night because there's a higher yield stone smelting recipe in FactoryMod because firstly it's not a replacement and secondly because the rates and costs are reasonable.

Convoy20 commented:
> Yeeaaah, I still don't get why adding a bunch of FactoryMod recipes is preferable to adding pipes if your sole (or at least primary) argument is about TPS. (And before you try this again: I wont benefit from pipes) And I even suggested having iced flowing water and bubble column water paths instead of hopper chains.

Doublechests will still tank performance. Pipes to large storage will still suck for performance. Minecart chests will still be used. Pipes don't solve it.

>you rejected my cost:yield suggestion
 
You didn't make a suggestion. 

> Don't think it was just an example, iirc it was an estimation

Sure, but not existing numbers.



> Well that depends on whether the recipes produce more output than they would with their equivalent crafting recipes. 

I would use this if it had a charcoal cost that was not ridiculous. So that's false.


>I've advocated against higher outputs because their stated intention (with the exception of the piston recipe) is to add automation, not add economies of scale. 

I'm happy to pay charcoal cost if you want to add economy of scale, but you again don't seem to be advocating for that either. Instead just quibbling because you hate any additional factorymod additions away from vanilla, or at least that's my perspective of it because otherwise your takes are so illogical.

>I accept that I'm probably naïve here and that 3-4 DCs of charcoal to process 78 DCs of material is too high.. 

When that is a daily occurrence and my alternative is to bot it for free. Yes. 

>but your proposed changes reducing it down to iirc like 1 DC is far too inexpensive. It's still FactoryMod.. you should still have to consider the costs of building, using, and maintaining the factory…

As I mentioned in DM, would you support the recipes as they are now, in their own special factory that costs a significant amount and has upkeep costs?

>and if you're casually handling 78 DCs of material of something, you can probably afford less than a full stack of compacted charcoal (produced from a bottable renewable resource) without much of an issue at all, which defeats the point.

Less than a CS of charcoal is my balanced costs. Daily. And that is something I agreed is fine. Still painful, but fine. Nearly 3CS for one operation, is not fine, and almost every large producer would just bot it instead.


> Sorry, but I'm not going to sit idly and smiley while you spread blatant lies about me.

I'm trying to understand your point here and it seems like you just dislike factorymod. Like I'm trying to justify what you say but your points are just so conflicting and confusing.
 
>stone smelting recipe in FactoryMod because firstly it's not a replacement and secondly because the rates and costs are reasonable.

Ever heard of a furnace? Because I certainly love them.

Quinndolin commented:
Some potential solutions:

1 - Little Nugget Big Factory
Nerf nugget output of portal farms by some factor
Buff nugget -> ingot recipe by that same factor
'Modest' charcoal cost

2 -  The Convoy Classic
nugget -> ingot recipe, same rate as vanilla
'Small' charcoal cost

3 -  Nugget? No Thank You
Remove nuggets from piglin drops and just make them drop ingots instead

Protonull commented:
> Doublechests will still tank performance. Pipes to large storage will still suck for performance. Minecart chests will still be used. Pipes don't solve it.

Don't understand this argument, firstly because performance issues with double chests originate from Citadel, which is something that can and should be fixed, not designed around. And secondly, I presume that you're using double chests in your storage and movements of things anyway so the point is moot.

----

> You didn't make a suggestion.

Not formally, but I made it pretty clear that I supported the original rates of.. what was it? 63 to 7? Figured you'd understand what I meant by "my suggestion"… guess not.

----

> Sure, but not existing numbers.

Literally irrelevant.

----

> I would use this if it had a charcoal cost that was not ridiculous. So that's false.

You misreading me again.. the "depends" is answering "will anyone use them?" not whether *you* will use them.

----

> Instead just quibbling because you hate any additional factorymod additions away from vanilla, or at least that's my perspective of it because otherwise your takes are so illogical.

Huh, TIL that I both simultaneously hate your proposals because I want everything FactoryMod to benefit me personally, and because I hate FactoryMod.

----

> When that is a daily occurrence and my alternative is to bot it for free. Yes.

And that'd be *your* choice, others may choose to absorb the cost ¯\_(ツ)_/¯

----

> As I mentioned in DM, would you support the recipes as they are now, in their own special factory that costs a significant amount and has upkeep costs?

Probably not because my objection to your now committed cost:yield proposals isn't about *where* the costs are, but the costs themselves. I don't necessarily mind going lower than 3-4 DCs, but your less than 1 DC to handle 78 DCs worth of materials, or 3,174-ish (if my math is correct) runs of the factory. The only precedent for such low costs is the Compactor, but that's only because the consumed cost is the Crate, not the charcoal.

----

> I'm trying to understand your point here and it seems like you just dislike factorymod.

It's pretty easy to understand… you made up lies about me, such as me "hating FactoryMod", and I'm refuting these lies. It's not complicated.

Protonull commented:
> Some potential solutions:
> 
> 1 - Little Nugget Big Factory
> Nerf nugget output of portal farms by some factor
> Buff nugget -> ingot recipe by that same factor (or slightly more !)
> 'Modest' charcoal cost
> 
> 2 - The Convoy Classic
> nugget -> ingot recipe, same rate as vanilla
> 'Small' charcoal cost
> 
> 3 - Nugget? No Thank You
> Remove nuggets from piglin drops and just make them drop ingots instead
> 
> I quite like door Number 3 but I find all of them agreeable.

tbh, I also prefer Option 3 because it's the simplest and likely easiest to balance, though we'd need to do some number crunching to result in a similar-ish yield of ingots so that it's not just a straight up buff to portal farm yields.

Quinndolin commented:
> 
> 
> > Some potential solutions:
> > 1 - Little Nugget Big Factory
> > Nerf nugget output of portal farms by some factor
> > Buff nugget -> ingot recipe by that same factor (or slightly more !)
> > 'Modest' charcoal cost
> > 2 - The Convoy Classic
> > nugget -> ingot recipe, same rate as vanilla
> > 'Small' charcoal cost
> > 3 - Nugget? No Thank You
> > Remove nuggets from piglin drops and just make them drop ingots instead
> > I quite like door Number 3 but I find all of them agreeable.
> 
> tbh, I also prefer Option 3 because it's the simplest and likely easiest to balance, though we'd need to do some number crunching to result in a similar-ish yield of ingots so that it's not just a straight up buff to portal farm yields.

What is it about FM that makes you want the increased coal costs compared to this PR versus having it be completely automatic and free with no set up cost

Protonull commented:
> > > Some potential solutions:
> > > 1 - Little Nugget Big Factory
> > > Nerf nugget output of portal farms by some factor
> > > Buff nugget -> ingot recipe by that same factor (or slightly more !)
> > > 'Modest' charcoal cost
> > > 2 - The Convoy Classic
> > > nugget -> ingot recipe, same rate as vanilla
> > > 'Small' charcoal cost
> > > 3 - Nugget? No Thank You
> > > Remove nuggets from piglin drops and just make them drop ingots instead
> > > I quite like door Number 3 but I find all of them agreeable.
> > 
> > 
> > tbh, I also prefer Option 3 because it's the simplest and likely easiest to balance, though we'd need to do some number crunching to result in a similar-ish yield of ingots so that it's not just a straight up buff to portal farm yields.
> 
> What is it about FM that makes you want the increased coal costs compared to this PR versus having it be completely automatic and free with no set up cost

Because I'm not "increasing" charcoal costs versus this PR.. Bonkill amended the PR *after* we had our discussion in VC.. I haven't increased charcoal costs versus the PR, he's lowered the charcoal costs of the PR.

Quinndolin commented:
> 
> 
> > > > Some potential solutions:
> > > > 1 - Little Nugget Big Factory
> > > > Nerf nugget output of portal farms by some factor
> > > > Buff nugget -> ingot recipe by that same factor (or slightly more !)
> > > > 'Modest' charcoal cost
> > > > 2 - The Convoy Classic
> > > > nugget -> ingot recipe, same rate as vanilla
> > > > 'Small' charcoal cost
> > > > 3 - Nugget? No Thank You
> > > > Remove nuggets from piglin drops and just make them drop ingots instead
> > > > I quite like door Number 3 but I find all of them agreeable.
> > > 
> > > 
> > > tbh, I also prefer Option 3 because it's the simplest and likely easiest to balance, though we'd need to do some number crunching to result in a similar-ish yield of ingots so that it's not just a straight up buff to portal farm yields.
> > 
> > 
> > What is it about FM that makes you want the increased coal costs compared to this PR versus having it be completely automatic and free with no set up cost
> 
> Because I'm not "increasing" charcoal costs versus this PR.. Bonkill amended the PR _after_ we had our discussion in VC.. I haven't increased charcoal costs versus the PR, he's lowered the charcoal costs of the PR.

What? That doesn't matter. Regardless of whether they're being increased un-increased etc whatever. You're now saying you prefer a solution that has NO charcoal cost whereas previously charcoal costs seemed like a dealbreaker for you.

Protonull commented:
> You're now saying you prefer a solution that has NO charcoal cost whereas previously charcoal costs seemed like a dealbreaker for you.

Ooooh, oke, I get what you're saying now.

Starting off, having all this be a part of FactoryMod is imho not ideal because, at least in my own personal opinion, FactoryMod is not *about* automation, it's about economies of scale and encouraging cooperation.. like you *can* automate FactoryMod, but that's not *the point*, again, at least in my opinion. These proposed changes however are to render bot-crafting redundant, to remove the need for bots by automating the crafting of items via FactoryMod for the purposes of farming on an enormous scale, hence why there's a gold nugget to gold ingot recipe but not an iron nugget to iron ingot recipe for example.

I'd personally prefer for there to be a Tekkit-style solution where there's an auto-crafting bench, but given how that doesn't exist on civ (at least not yet), then I concede the only real option is to shoehorn it into FactoryMod.. but if it's going in FactoryMod then it should have the same costs as FactoryMod. The follow-up argument to that is "If it has the same costs, why not the same benefits?" Well, not all FactoryMod recipes produce higher outputs despite having a cost, for example: cleaning red sand, dirt->grass, dirt->podzol, dirt->mycelium, converting sapling types, shaving snow, etc.

So the reason why I prefer Option 3, of amending Piglin drops, is because then it's not shoehorning in a bunch of automation recipes into FactoryMod, especially given that already does too many things (which is not the same thing as saying that FactoryMod should only be for XP), and because it'll be easier to maintain.

wingzero54 commented:
I'm hesitant because I don't really see the improvement you claim. It would still require a lot of hoppers, and a reduction of like 10 or 20 hoppers with this change doesn't seem like it would make that much of an impact on performance. I would like to see a before-after comparison of what a big farm would look like, and the performance difference of both

Convoy20 commented:
> I'm hesitant because I don't really see the improvement you claim. It would still require a lot of hoppers, and a reduction of like 10 or 20 hoppers with this change doesn't seem like it would make that much of an impact on performance. I would like to see a before-after comparison of what a big farm would look like, and the performance difference of both

These changes matched with pony's input/output changes if merged would be a drastic difference. 

This is one of my WIP designs for a high output farm: https://i.imgur.com/VHLFkkc.png

Right now it is eight overflow splitters, currently matched with 4 minecart chests on top of another hopper leading to a doublechest. A total of 24 hoppers, and 32 minecart chests. For a fully operational farm I would need around 40 minecart chests per station, for a total of 320 minecart chests.

For current farms Minecart chests are .66% of the tick and hoppers are 1.93% of the tick from an old timings I looked up. Encouraging more hopper or minecart chest use will just allow those %'s to grow. 

The replacement for this would be one hopper, a double (or triple) factory, which both crafts gold ingots and blocks as well as compacts the gold blocks.

TheOrangeWizard commented:
fully support these additions alongside https://github.com/CivClassic/FactoryMod/pull/58

can't comment on performance issues but i think it would reduce reliance on crafting/inventory bots for scale production & make for less painful gameply
